packages/mobile/src/screens/favorites-screen/FavoritesDownloadSection.tsx

```diff
@@ -1,27 +1,55 @@
-import { useState } from 'react'
+import { useCallback, useState } from 'react'
 
 import { View } from 'react-native'
+import { TouchableOpacity } from 'react-native-gesture-handler'
+import { useDispatch, useSelector } from 'react-redux'
 
+import { setVisibility } from 'app/store/drawers/slice'
+import { DOWNLOAD_REASON_FAVORITES } from 'app/store/offline-downloads/constants'
+import { getCollectionDownloadStatus } from 'app/store/offline-downloads/selectors'
 import { makeStyles } from 'app/styles'
 
-import { DownloadFavoritesSwitch } from './DownloadFavoritesSwitch'
 import { DownloadProgress } from './DownloadProgress'
 import { FavoritesDownloadStatusIndicator } from './FavoritesDownloadStatusIndicator'
 
 const useStyles = makeStyles(() => ({
   root: { flexDirection: 'row', alignItems: 'center' }
 }))
 
+const OfflineListeningIndicatorButton = () => {
+  const dispatch = useDispatch()
+  const isFavoritesMarkedForDownload = useSelector((state) =>
+    Boolean(getCollectionDownloadStatus(state, DOWNLOAD_REASON_FAVORITES))
+  )
+  const [switchValue, setSwitchValue] = useState(isFavoritesMarkedForDownload)
+
+  const handlePressStatusIndicator = useCallback(() => {
+    dispatch(
+      setVisibility({
+        drawer: 'OfflineListening',
+        visible: true,
+        data: {
+          isFavoritesMarkedForDownload: switchValue,
+          onSaveChanges: setSwitchValue
+        }
+      })
+    )
+  }, [dispatch, switchValue])
+
+  return (
+    <TouchableOpacity onPress={handlePressStatusIndicator}>
```

Comment:
it might not work, but i wonder if we could use IconButton for this? the nice thing with that is it comes in with some built-in hitSlop to make it easier to press.

Reply:
sadly we cannot because the "loading" indicator state is a rive component not an icon component. can add in a hitslop here if we run into issues pressing the button in qa!

Reply:
nice yeah makes sense, and sounds like a plan :)